magento/magento2
# Fixed issue #14550

### Description (*)

Price Filter Issue in Search criteria not returning correct product #14550
### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2 #14550

### Manual testing scenarios (*)

Hit the products REST api using any client with searchCriteria Category id as 23 and price filter as price between $40 and $50 or between $70 and $80. Here is the URL that i built
rest/V1/products?searchCriteria[filter_groups][1][filters][0][field]=category_id&searchCriteria[filter_groups][1][filters][0][value]=23&searchCriteria[filter_groups][1][filters][0][condition_type]=eq&searchCriteria[filter_groups][2][filters][0][field]=price&searchCriteria[filter_groups][2][filters][0][value]=40&searchCriteria[filter_groups][2][filters][0][condition_type]=from&searchCriteria[filter_groups][3][filters][0][field]=price&searchCriteria[filter_groups][3][filters][0][value]=50&searchCriteria[filter_groups][3][filters][0][condition_type]=to&searchCriteria[filter_groups][2][filters][1][field]=price&searchCriteria[filter_groups][2][filters][1][value]=70&searchCriteria[filter_groups][2][filters][1][condition_type]=from&searchCriteria[filter_groups][3][filters][1][field]=price&searchCriteria[filter_groups][3][filters][1][value]=80&searchCriteria[filter_groups][3][filters][1][condition_type]=to

Check the response of the Call

### Questions or comments


### Contribution checklist (*)
